src/test/feature/lib/file_replace.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <fstream>
+
+#include "file_replace.h"
+
+using std::string;
+using std::map;
+using std::endl;
+using std::ifstream;
+using std::ofstream;
+
+namespace hawq {
+namespace test {
+
+string FileReplace::replaceAllOccurrences(
+	string str,
+	const string& src,
+	const string& dst)
+{
+	size_t start_pos = 0;
+	while ((start_pos = str.find(src, start_pos)) != string::npos)
+	{
+		str.replace(start_pos, src.length(), dst);
+		start_pos += dst.length();
+	}
+
+	return str;
+}
+
+void FileReplace::replace(
+	const string& file_src,
+	const string& file_dst,
+	const map<string, string>& strs_src_dst)
```

Comment:
It is better to use unordered_map instead of map.
For adding element in unordered_map or map, I would prefer insert instead of operator [] since insert would do a check if duplicate keys are going to be added.